sstables/sstables.cc

```diff
@@ -1097,28 +1097,32 @@ future<> sstable::read_simple(T& component, const io_priority_class& pc) {
 
     auto file_path = filename(Type);
     sstlog.debug(("Reading " + sstable_version_constants::get_component_map(_version).at(Type) + " file {} ").c_str(), file_path);
-    return new_sstable_component_file(_read_error_handler, Type, open_flags::ro).then([this, &component] (file fi) {
-        auto fut = fi.size();
-        return fut.then([this, &component, fi = std::move(fi)] (uint64_t size) {
-            auto r = make_lw_shared<file_random_access_reader>(std::move(fi), size, sstable_buffer_size);
-            auto fut = parse(*_schema, _version, *r, component);
-            return fut.finally([r] {
-                return r->close();
-            }).then([r] {});
-        });
-    }).then_wrapped([file_path] (future<> f) {
         try {
-            f.get();
-        } catch (std::system_error& e) {
+
+        file fi = co_await new_sstable_component_file(_read_error_handler, Type, open_flags::ro);
+        uint64_t size = co_await fi.size();
+
+        auto r = make_lw_shared<file_random_access_reader>(std::move(fi), size, sstable_buffer_size);
```

Comment:
It looks like `r` doesn't need to be lw_shared_ptr -- it doesn't outsurvive this function, and `parse()` accepts it by reference, now by lw-pointer

Reply:
you're right.